### PR TITLE
feat: get single blogpost

### PR DESCRIPTION
**Pull request by RB-Isiaq**
## What does this PR do?

This PR is all about fetching a single blog post.

## How should this be manually tested?

Send a GET request to `/api/v1/biogs/:id` to get a single blog post.

## Checklist of what you did

- [x] Implemented endpoint to get a single blog post  by id.
- [X] Returned blog post data including: Title, Content, Publish Date, Images, tags Author.
- [x] Returned success message with the right response payload.
- [x] Added unit tests for the get single blog service

## Response

Status: 200 OK
Body:
``` json 
{    
     "status": 200,
     "message": "message",
     "data": {
        "blog_id": "blog-id",
        "title": "Blog Post Title",
        "content": "Full content of the blog post...",
        "published_date": "2024-07-17T00:00:00Z",
        "author": "Author Name",
        "tags": ["tag1", "tag2" ],
        "image_urls": ["Image URL", "Image2_URL"]
    }
}
```
### Error Response:

Status: 500 Internal Server Error
Body:
Status: 200 OK
Body:
``` json 
{    
     "status": 500,
     "message": "message",
}
```
Not Found Response:
Status: 404 Not Found
Body:
``` json 
{    
     "status": 404,
     "message": "message",
}
```

Bad Request Response:

Status: 400 Bad Request
Body:
``` json 
{    
     "status": 400,
     "message": "message",
}
```

## Screenshots

image

## Linked Issue
#187

### CODE REVIEW

**Review by Cobidex — DISMISSED** [review]

**Review by davidolanrewaju — DISMISSED** [review]

File: src/modules/blogs/blogs.controller.ts [hunk 1]
```diff
@@ -11,11 +11,19 @@ export class BlogController {
   constructor(private readonly blogService: BlogService) {}
 
   @Post()
-  @UseGuards(SuperAdminGuard)
+  // @UseGuards(SuperAdminGuard)
```
Comment (Cobidex) [inline src/modules/blogs/blogs.controller.ts:14]: Why comment this out. If the guard is not needed. Please remove entirely

File: src/modules/blogs/blogs.controller.ts [hunk 2]
```diff
@@ -11,11 +11,19 @@ export class BlogController {
...
   @ApiResponse({ status: 403, description: 'Forbidden.' })
   async createBlog(@Body() createBlogDto: CreateBlogDto, @Request() req): Promise<any> {
     return await this.blogService.createBlog(createBlogDto, req.user);
   }
+
+  @Get(':id')
+  @ApiOperation({ summary: 'Get a single blog' })
+  @ApiResponse({ status: 200, description: 'Blog fetched successfully.', type: BlogResponseDto })
+  @ApiResponse({ status: 404, description: 'Blog not found.' })
+  async getSingleBlog(@Param('id') id: string, @Request() req): Promise<any> {
```
Comment (Cobidex) [inline src/modules/blogs/blogs.controller.ts:26]: Add validation for UUID. parseUuidPipe

File: src/modules/blogs/blogs.service.ts [hunk 3]
```diff
@@ -46,4 +47,28 @@ export class BlogService {
       created_at: savedBlog.created_at,
     };
   }
+
+  async getSingleBlog(blogId: string, user: User): Promise<any> {
```
Comment (Cobidex) [inline src/modules/blogs/blogs.service.ts:51]: The promise should have a shape not just any

File: src/modules/blogs/blogs.controller.ts (hunk 1 above)
Comment (RB-Isiaq) [reply src/modules/blogs/blogs.controller.ts]: Done.

File: src/modules/blogs/blogs.controller.ts (hunk 2 above)
Comment (RB-Isiaq) [reply src/modules/blogs/blogs.controller.ts]: Done.

File: src/modules/blogs/blogs.service.ts (hunk 3 above)
Comment (RB-Isiaq) [reply src/modules/blogs/blogs.service.ts:51]: Done @Cobidex

**Comment by RB-Isiaq** [comment]
@Cobidex @davidolanrewaju 

Kindly review. Thanks.

**Comment by RB-Isiaq** [comment]
@TheCodeGhinux Please Review

**Review by TheCodeGhinux — APPROVED** [review]

**Pull request merged**